app/src/parser.py

```diff
@@ -154,7 +159,7 @@ def extract(self, sheets, config):
 
                 for _, row in df.iterrows():
                     branch = row["Branche"]
-                    if branch in branch_groups and branch_groups[branch] == "individual":
+                    if branch_groups.get(branch) == "individual":
```

Comment:
I recall having some issues when using get here. Can you run the app and try some filters to test it, just to be on the safe side?